tests/controllers/cleanup/delete_tokens_test.go

```diff
@@ -0,0 +1,165 @@
+package feature_test
+
+import (
+	"context"
+	"fmt"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/auth"
+	"github.com/rancher/rancher/pkg/multiclustermanager"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"github.com/rancher/rancher/tests/controllers/common"
+	"github.com/rancher/wrangler/v3/pkg/crd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"testing"
+	"time"
+)
+
+type AuthConfigSuite struct {
+	suite.Suite
+	ctx               context.Context
+	cancel            context.CancelFunc
+	testEnv           *envtest.Environment
+	managementContext *config.ManagementContext
+	wranglerContext   *wrangler.Context
+}
+
+func TestAuthConfigSuite(t *testing.T) {
+	suite.Run(t, new(AuthConfigSuite))
+}
+
+func (s *AuthConfigSuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithCancel(context.Background())
+
+	// Load CRD from YAML for REST Client
+	s.testEnv = &envtest.Environment{}
+	restCfg, err := s.testEnv.Start()
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), restCfg)
+
+	// Create CRDs
+	factory, err := crd.NewFactoryFromClient(restCfg)
+	assert.NoError(s.T(), err)
+
+	err = factory.BatchCreateCRDs(s.ctx,
+		crd.CRD{
+			SchemaObject: v3.Token{},
+			NonNamespace: true,
+		},
+		crd.CRD{
+			SchemaObject: v3.AuthConfig{},
+			NonNamespace: true,
+		},
+	).BatchWait()
+	assert.NoError(s.T(), err)
+
+	// Create the wrangler and management contexts
+	wranglerContext, err := wrangler.NewContext(s.ctx, nil, restCfg)
+	assert.NoError(s.T(), err)
+
+	scaledContext, clusterManager, _, err := multiclustermanager.BuildScaledContext(s.ctx, wranglerContext, &multiclustermanager.Options{})
+	assert.NoError(s.T(), err)
+	s.managementContext, err = scaledContext.NewManagementContext()
+	assert.NoError(s.T(), err)
+
+	// Register the auth controller
+	auth.RegisterEarly(s.ctx, s.managementContext, clusterManager)
+
+	// Start controllers
+	common.StartNormanControllers(s.ctx, s.T(), s.managementContext,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "AuthConfig",
+		},
+	)
+
+	// Start caches
+	common.StartWranglerCaches(s.ctx, s.T(), s.managementContext.Wrangler,
+		schema.GroupVersionKind{
+			Group:   "management.cattle.io",
+			Version: "v3",
+			Kind:    "Token",
+		},
+	)
+}
+
+func (s *AuthConfigSuite) TestTokensCleanup() {
+	t := assert.New(s.T())
```

Comment:
nit: I find it a bit jarring to read different uses of assert in different tests and subtests in close proximity:
- assert.X(s.T(), ...)
- t.X()
- assert.X(c, ...)

Please feel free to ignore but personally I'd probably stick with something more explicit and easy to follow by just glancing at it.
```
t := s.T()
...
assert.X(t, ...)
require.X(t, ...)
```

Reply:
Yes, not a NIT. It was a bit confusing also for me. I'll fix it!